### PR TITLE
Jdk 11

### PR DESCRIPTION
**Pull request by teams2ua**
Update djinni sbt + scala to work woth newer JDK
Update JDK to have a package on debian buster
Update debian to Buster to have a newer gcc
Update gcc to have C++17

### CODE REVIEW

File: .circleci/build_lib.sh [hunk 1]
```diff
@@ -101,7 +101,7 @@ fi
 
 }
 
-export JAVA_HOME="/usr/lib/jvm/java-8-openjdk"
+export JAVA_HOME="/usr/lib/jvm/java-11-openjdk-amd64"
```
Comment (phaazon) [inline .circleci/build_lib.sh]: That `amd64` brings concerns. Why are we now platform-dependent?

File: core/src/crypto/AESCipher.cpp [hunk 2]
```diff
@@ -57,7 +57,6 @@ namespace ledger {
 
                 // Create an IVt
                 auto IV = _rng->getRandomBytes(AES256::BLOCK_SIZE);
-                assert(IV.size() == AES256::BLOCK_SIZE);
```
Comment (phaazon) [inline core/src/crypto/AESCipher.cpp]: Nope. You can’t do that. 🙅‍♂️ That assert is needed.

Comment (teams2ua) [reply core/src/crypto/AESCipher.cpp]: ok, put it back

File: .circleci/build_lib.sh (hunk 1 above)
Comment (teams2ua) [reply .circleci/build_lib.sh]: Don't know what to do here, this is the path the package (apt-get install openjdk-11-jdk) is installed... What would you recommend ?

File: core/src/crypto/AESCipher.cpp (hunk 2 above)
Comment (phaazon) [reply core/src/crypto/AESCipher.cpp]: 🔞

**Pull request merged**